resolwe_bio/processes/import_data/proteomics_data.py

```diff
@@ -0,0 +1,221 @@
+"""Upload proteomics data."""
+import re
+from pathlib import Path
+
+import pandas as pd
+
+from resolwe.process import FileField, Process, SchedulingClass, StringField
+
+
+def change_suffix(path):
+    """Change suffix of a file to lowercase."""
+    new_path = path.with_suffix(path.suffix.lower())
+    path.replace(new_path)
+    return new_path
+
+
+def prepare_filename(fname):
+    """Return a sanitized string that can be used as a file name."""
+    return re.sub(r"(?u)[^-\w.]", "", str(fname).strip().replace(" ", "_"))
+
+
+class UploadProteomicsData(Process):
+    """Upload a mass spectrometry proteomics sample data file.
+
+    The input 5-column tab-delimited file with the .txt suffix is
+    expected to contain a header line with the following meta-data
+    column names: "Uniprot ID", "Gene symbol", "Protein name" and
+    "Number of peptides". The fifth column contains the sample data.
+    """
+
+    slug = "upload-proteomics-sample"
+    name = "Upload proteomics sample"
+    process_type = "data:proteomics:massspectrometry"
+    version = "1.0.0"
+    category = "Import"
+    scheduling_class = SchedulingClass.BATCH
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {"docker": {"image": "resolwebio/common:1.6.0"}},
+        "resources": {"cores": 1, "memory": 2048},
+    }
+    entity = {
+        "type": "sample",
+        "descriptor_schema": "sample",
+    }
+    data_name = '{{ src.file|default("?") }}'
+
+    class Input:
+        """Input field to process UploadProteomicsData."""
+
+        src = FileField(label="Table containing mass spectrometry data (.txt)")
+        species = StringField(
+            label="Species",
+            description="Select a species name from the dropdown menu "
+            "or write a custom species name in the species field.",
+            allow_custom_choice=True,
+            choices=[
+                ("Homo sapiens", "Homo sapiens"),
+                ("Mus musculus", "Mus musculus"),
+                ("Rattus norvegicus", "Rattus norvegicus"),
+            ],
+        )
+        source = StringField(
+            label="Protein ID database source",
+            default="UniProtKB",
+            choices=[("UniProtKB", "UniProtKB")],
+        )
+
+    class Output:
+        """Output field of the process UploadProteomicsData."""
+
+        table = FileField(label="Uploaded table")
+        species = StringField(label="Species")
+        source = StringField(label="Source")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+
+        table_path = inputs.src.import_file(imported_format="extracted")
+
+        extensions = [".txt"]
+        path = Path(table_path)
+
+        if path.suffix in [e.upper() for e in extensions]:
```

Comment:
Is this something common? I haven't seen much files of the form `filename.TXT`...